### PR TITLE
[REVIEW] Update/fix doxygen comments for the csv_read_arg struct

### LINKED ISSUE #695: [DOC] Missing Doxygen comments for completed read_csv() features in io_types.h
List of implemented parameters that are missing Doxygen comments:
- usecol
- nrows
- header
- dayfirst

### PR DESCRIPTION
**Pull request by vuule**
Update the csv_read_arg definition to match the doxygen format used in the project.

Fixes #695

### CODE REVIEW

**Comment by vuule** [comment]
rerun tests

**Comment by vuule** [comment]
rerun tests

**Comment by vuule** [comment]
rerun tests

**Comment by vuule** [comment]
rerun tests

**Comment by vuule** [comment]
rerun tests

File: cpp/include/cudf/io_types.h [hunk 1]
```diff
@@ -27,77 +27,84 @@
...
-  bool          delim_whitespace;           /**< use white space as the delimiter - default is false.  This overrides the delimiter argument    */
-  bool          skipinitialspace;           /**< skip white spaces after the delimiter - default is false                                       */
+  gdf_csv_input_form	input_data_form;	///< Type of source of CSV data
+  const char			*filepath_or_buffer;///< If input_data_form is FILE_PATH, contains the filepath. If input_data_type is HOST_BUFFER, points to the host memory buffer
+  size_t				buffer_size;		///< If input_data_form is HOST_BUFFER, represents the size of the buffer in bytes. Unused otherwise
 
+  bool          windowslinetermination;     ///< States if we should \r\n as our line termination
+  char          lineterminator;             ///< define the line terminator character. Default is  '\n'
+  char          delimiter;                  ///< define the field separator, default is ',' This argument is also called 'sep' 
+  bool          delim_whitespace;           //< Use white space as the delimiter - default is false. This overrides the delimiter argument
```
Comment (harrism) [inline cpp/include/cudf/io_types.h]:
```suggestion
  bool          delim_whitespace;           ///< Use white space as the delimiter - default is false. This overrides the delimiter argument
```

File: cpp/include/cudf/io_types.h [hunk 2]
```diff
@@ -27,77 +27,84 @@
...
-  size_t				buffer_size;		/**< If input_data_form is HOST_BUFFER, represents the size of the buffer in bytes. Unused otherwise */
-
-  bool          windowslinetermination;     /**< States if we should \r\n as our line termination>**/
-  char          lineterminator;             /**< define the line terminator character.  Default is  '\n'                                        */
-  char          delimiter;                  /**< define the field separator, default is ','   This argument is also called 'sep'                */
-  bool          delim_whitespace;           /**< use white space as the delimiter - default is false.  This overrides the delimiter argument    */
-  bool          skipinitialspace;           /**< skip white spaces after the delimiter - default is false                                       */
+  gdf_csv_input_form	input_data_form;	///< Type of source of CSV data
+  const char			*filepath_or_buffer;///< If input_data_form is FILE_PATH, contains the filepath. If input_data_type is HOST_BUFFER, points to the host memory buffer
```
Comment (harrism) [inline cpp/include/cudf/io_types.h]:
```suggestion
  const char			*filepath_or_buffer; ///< If input_data_form is FILE_PATH, contains the filepath. If input_data_type is HOST_BUFFER, points to the host memory buffer
```

File: cpp/include/cudf/io_types.h [hunk 3]
```diff
@@ -27,77 +27,84 @@
...
+  const char    **use_cols_char;            ///< Names of columns to be returned. CSV reader will only process those columns, another read is needed to get full data
+  int           use_cols_char_len;          ///< Number of elements in use_cols_char_len
 
-  gdf_size_type skiprows;                   /**< number of rows at the start of the files to skip, default is 0                                 */
-  gdf_size_type skipfooter;                 /**< number of rows at the bottom of the file to skip - default is 0                                */
+  gdf_size_type skiprows;                   ///< Number of rows at the start of the files to skip, default is 0
+  gdf_size_type skipfooter;                 ///< Number of rows at the bottom of the file to skip - default is 0
 
-  bool          skip_blank_lines;           // whether or not to ignore blank lines
+  bool          skip_blank_lines;           // Whether or not to ignore blank lines
```
Comment (harrism) [inline cpp/include/cudf/io_types.h]:
```suggestion
  bool          skip_blank_lines;           ///* Whether or not to ignore blank lines
```

File: cpp/include/cudf/io_types.h [hunk 4]
```diff
@@ -27,77 +27,84 @@
...
-  int           num_true_values;            /**< number of values in the true_values list */
-  const char    **false_values;             /**< list of values to recognize as boolean False */
-  int           num_false_values;           /**< number of values in the true_values list */
+  const char    **true_values;              ///< List of values to recognize as boolean True
+  int           num_true_values;            ///< Number of values in the true_values list
+  const char    **false_values;             ///< List of values to recognize as boolean False
+  int           num_false_values;           ///< Number of values in the true_values list
 
-  const char    **na_values;                // array of char *    what should be considered as True - each char string contains {col ID, na value, ...} - this will allow multiple na values to be specified over multiple columns
+  const char    **na_values;                // Array of strings that should be considered as NA
```
Comment (harrism) [inline cpp/include/cudf/io_types.h]:
```suggestion
  const char    **na_values;                ///< Array of strings that should be considered as NA
```

File: cpp/include/cudf/io_types.h [hunk 5]
```diff
@@ -27,77 +27,84 @@
...
 
-  char          *prefix;                    // if there is no header or names, append this to the column ID as the name
-  bool          mangle_dupe_cols;           // if true: duplicate columns will be specified as (deleted because utf-8 chars kill the build)
+  char          *prefix;                    // If there is no header or names, append this to the column ID as the name
```
Comment (harrism) [inline cpp/include/cudf/io_types.h]:
```suggestion
  char          *prefix;                    ///< If there is no header or names, append this to the column ID as the name
```

File: cpp/include/cudf/io_types.h [hunk 6]
```diff
@@ -27,77 +27,84 @@
...
+  bool          mangle_dupe_cols;           ///< If true, duplicate columns get a suffix. If false, data will be overwritten if there are columns with duplicate names
 
-  bool          parse_dates;                // parse date field into date32 or date64.  If false then date fields are saved as a string. Specifying a date dtype overrides this
-  bool          infer_datetime_format;      // try and determine the date format
-  bool          dayfirst;                   // is the first value the day?  DD/MM  versus MM/DD
+  bool          parse_dates;                // Parse date field into date32 or date64.  If false then date fields are saved as a string. Specifying a date dtype overrides this
```
Comment (harrism) [inline cpp/include/cudf/io_types.h]:
```suggestion
  bool          parse_dates;                ///< Parse date field into date32 or date64.  If false then date fields are saved as a string. Specifying a date dtype overrides this
```

File: cpp/include/cudf/io_types.h [hunk 7]
```diff
@@ -27,77 +27,84 @@
...
+  bool          infer_datetime_format;      // Try and determine the date format
```
Comment (harrism) [inline cpp/include/cudf/io_types.h]:
```suggestion
  bool          infer_datetime_format;      ///< Try and determine the date format
```

File: cpp/include/cudf/io_types.h [hunk 8]
```diff
@@ -27,77 +27,84 @@
...
 
-  char          quotechar;                  /**< define the character used to denote start and end of a quoted item                             */
-  bool          quoting;                    /**< treat string fields as quoted item and remove the first and last quotechar                     */
-  bool          doublequote;                /**< indicates whether to interpret two consecutive quotechar inside a field as a single quotechar  */
+  char          quotechar;                  ///< Define the character used to denote start and end of a quoted item
+  bool          quoting;                    ///< Treat string fields as quoted item and remove the first and last quotechar
+  bool          doublequote;                ///< Indicates whether to interpret two consecutive quotechar inside a field as a single quotechar
 
-  char          escapechar;                 // single char    - char used as the escape character
+  char          escapechar;                 // Single character used as the escape character
```
Comment (harrism) [inline cpp/include/cudf/io_types.h]:
```suggestion
  char          escapechar;                 ///< Single character used as the escape character
```

File: cpp/include/cudf/io_types.h [hunk 9]
```diff
@@ -27,77 +27,84 @@
...
 
-  char          comment;                    // single char    - treat line or remainder of line as an unprocessed comment
+  char          comment;                    // Single character indicating that the remainder of line is a comment
```
Comment (harrism) [inline cpp/include/cudf/io_types.h]:
```suggestion
  char          comment;                    ///< Single character indicating that the remainder of line is a comment
```

**Comment by vuule** [comment]
@harrism Thank you for the review. I committed several suggested changes.

Regarding the remaining suggestions: the members that don't have doxygen style comments are left like that on purpose - these are the parameters that we don't support yet (support incoming for several of them). We plan to update the comment style in PRs that add the functionality.

**Comment by harrism** [comment]
Ah, my mistake. Please resolve any comments that don't apply.

**Comment by vuule** [comment]
> Ah, my mistake. Please resolve any comments that don't apply.

Resolved.

**Pull request merged**